test/js/node/http2/node-http2.test.js

```diff
@@ -1334,3 +1329,581 @@ it("sensitive headers should work", async () => {
+      session.on("stream", stream => {
+        stream.on("end", function () {
+          this.respond(
+            {
+              ":status": 200,
+            },
+            {
+              endStream: true,
+            },
+          );
+        });
+        stream.resume();
+      });
+    });
+
+    server.listen(0, () => {
+      const port = server.address().port;
+      const client = http2.connect(`http://localhost:${port}`);
+
+      function next(i) {
+        if (i === 10000) {
+          client.close();
+          server.close();
+          resolve();
+          return;
+        }
+
+        const stream = client.request({ ":method": "POST" });
+
+        stream.on("response", function (headers) {
+          expect(headers[":status"]).toBe(200);
+
+          this.on("close", () => next(i + 1));
+        });
+
+        stream.end();
+      }
+
+      // Start the sequence with the first request
+      next(0);
+    });
+  });
+}, 10_000);
+
+it("http2.createServer validates input options", () => {
+  // Test invalid options passed to createServer
+  const invalidOptions = [1, true, "test", null, Symbol("test")];
+
+  invalidOptions.forEach(invalidOption => {
+    expect(() => http2.createServer(invalidOption)).toThrow(
+      'The "options" argument must be of type Object.' + invalidArgTypeHelper(invalidOption),
+    );
+  });
+
+  // Test invalid options.settings passed to createServer
+  invalidOptions.forEach(invalidSettingsOption => {
+    expect(() => http2.createServer({ settings: invalidSettingsOption })).toThrow(
+      'The "options.settings" property must be of type Object.' + invalidArgTypeHelper(invalidSettingsOption),
+    );
+  });
+
+  // Test that http2.createServer validates numeric range options
+  const rangeTests = {
+    maxSessionInvalidFrames: [
+      {
+        val: -1,
+        err: {
+          name: "RangeError",
+          code: "ERR_OUT_OF_RANGE",
+        },
+      },
+      {
+        val: Number.NEGATIVE_INFINITY,
+        err: {
+          name: "RangeError",
+          code: "ERR_OUT_OF_RANGE",
+        },
+      },
+    ],
+    maxSessionRejectedStreams: [
+      {
+        val: -1,
+        err: {
+          name: "RangeError",
+          code: "ERR_OUT_OF_RANGE",
+        },
+      },
+      {
+        val: Number.NEGATIVE_INFINITY,
+        err: {
+          name: "RangeError",
+          code: "ERR_OUT_OF_RANGE",
+        },
+      },
+    ],
+  };
+
+  Object.entries(rangeTests).forEach(([opt, tests]) => {
+    tests.forEach(({ val, err }) => {
+      expect(() => http2.createServer({ [opt]: val })).toThrow();
+
+      // Note: Bun's expect doesn't have the same detailed error matching as Node's assert,
+      // so we're just checking that it throws an error with the expected name
+      try {
+        http2.createServer({ [opt]: val });
+      } catch (error) {
+        expect(error.name).toBe(err.name);
+        expect(error.code).toBe(err.code);
+      }
```

Comment:
```suggestion
      let error;

      try {
        http2.createServer({ [opt]: val });
      } catch (e) {
        error = e;
      }
      
      expect(error).toBeTruthy();
      expect(error?.name).toBe(err.name);
      expect(error?.code).toBe(err.code);
```

Otherwise even if createServer doesn't throw, the test still passes